common/scala/src/main/scala/whisk/common/RingBuffer.scala

```diff
@@ -0,0 +1,16 @@
+package whisk.common
+
+import org.apache.commons.collections.buffer.CircularFifoBuffer
+
+object RingBuffer {
+    def apply[T](size: Int) = new RingBuffer[T](size)
+}
+
+class RingBuffer[T](size: Int) {
+    private val inner = new CircularFifoBuffer(size)
+
+    def add(el: T) = inner.add(el)
+    def count(el: T) = {
+        inner.toArray().asInstanceOf[Array[T]].count(_ == el)
+    }
```

Comment:
My initial implementation had just `add` and `toList` which makes it easier to apply whatever functionality i need to the values in the buffer outside of the buffer's business code. This might be okay, if we just need `count`.